src/components/ImportForm/GitImportForm.tsx

```diff
@@ -129,7 +130,11 @@ const GitImportForm: React.FunctionComponent<GitImportFormProps> = ({
 
   const handleReset = () => {
     track(TrackEvents.ButtonClicked, { link_name: 'import-leave', workspace });
-    navigate(-1);
+    showModal<{ leave: boolean }>(createCloseImportFormModal()).closed.then(({ leave }) => {
```

Comment:
I think before showing the modal we should also check if the form is dirty or not and only show the modal if it is.